hal/stm32l4.c

```diff
@@ -0,0 +1,262 @@
+ *
+ * wolfBoot is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfBoot is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#include <stdint.h>
+#include <image.h>
+#include "stm32l4xx_hal.h"
+
+/* Assembly helpers */
+#define DMB() asm volatile ("dmb")
+
+/*** RCC ***/
+
+#define RCC_PRESCALER_DIV_NONE 0
+
+
+
+static uint32_t Address = 0, PAGEError = 0;
+
+
+static FLASH_EraseInitTypeDef EraseInitStruct;
+  
+  
+static uint32_t RAMFUNCTION GetPage(uint32_t Addr)
+{
+    uint32_t page = 0;
+  
+    if (Addr < (FLASH_BASE + FLASH_BANK_SIZE))
+        page = (Addr - FLASH_BASE) / FLASH_PAGE_SIZE;
+    else
+        page = (Addr - (FLASH_BASE + FLASH_BANK_SIZE)) / FLASH_PAGE_SIZE;
+  
+    return page;
+}
+
+
+static uint32_t RAMFUNCTION GetBank(uint32_t Addr)
+{
+    uint32_t bank = 0;
+  
+    if (READ_BIT(SYSCFG->MEMRMP, SYSCFG_MEMRMP_FB_MODE) == 0) {
+        if (Addr < (FLASH_BASE + FLASH_BANK_SIZE)) {
+            bank = FLASH_BANK_1;
+        } else {
+            bank = FLASH_BANK_2;
+        }
+    } else {
+        if (Addr < (FLASH_BASE + FLASH_BANK_SIZE)) {
+            bank = FLASH_BANK_2;
+        } else {
+            bank = FLASH_BANK_1;
+        }
+    }
+	
+    return bank;
+}
+
+static void RAMFUNCTION hal_flash_clear_errors(void)
+{
+     __HAL_FLASH_CLEAR_FLAG(FLASH_FLAG_ALL_ERRORS);
+}
+
+void RAMFUNCTION hal_flash_unlock(void)
+{
+     HAL_FLASH_Unlock();
+}
+
+void RAMFUNCTION hal_flash_lock(void)
+{
+    HAL_FLASH_Lock();
+}
+
+int RAMFUNCTION hal_flash_erase(uint32_t address,int len)
+{
+    uint32_t FirstPage = 0, NbOfPages = 0, BankNumber = 0;
+    uint32_t PAGEError = 0;
+    uint32_t end_address;
+    hal_flash_clear_errors();
+
+    if (len == 0) {
+        return -1;
+    }
+    hal_flash_unlock();
+
+    __HAL_FLASH_CLEAR_FLAG(FLASH_FLAG_OPTVERR);
+	
+    FirstPage = GetPage((uint32_t) address);
+    NbOfPages = GetPage((uint32_t) address) - FirstPage + 1;
+    BankNumber = GetBank((uint32_t) address);
+	
+    EraseInitStruct.TypeErase   = FLASH_TYPEERASE_PAGES;
+    EraseInitStruct.Banks       = BankNumber;
+    EraseInitStruct.Page        = FirstPage;
+    EraseInitStruct.NbPages     = NbOfPages;
+	
+    end_address = address + len - 1;
+	
+    for (uint32_t p = address; p < end_address; p += FLASH_PAGE_SIZE) {
+	    
+        if (HAL_FLASHEx_Erase(&EraseInitStruct, &PAGEError) != HAL_OK) {
+            break;
+	}
```

Comment:
Please check indent here. This file should use 4 spaces not tabs.

Reply:
There was indeed a tab at line 119